main/src/ca/uwaterloo/flix/language/errors/TypeError.scala

```diff
@@ -274,4 +274,24 @@ object TypeError {
     }
   }
 
+  /**
+    * No matching instance error.
+    *
+    * @param clazz the class of the instance.
+    * @param tpe   the type of the instance.
+    * @param loc   the location where the error occurred.
+    */
+  case class NoMatchingInstance(clazz: Symbol.ClassSym, tpe: Type, loc: SourceLocation) extends TypeError {
+    def summary: String = s"No instance of class '$clazz' for type '${FormatType.formatType(tpe)}'."
+
+    def message: VirtualTerminal = {
+      val vt = new VirtualTerminal()
+      vt << Line(kind, source.format) << NewLine
+      vt << ">> No instance of class '" << Red(clazz.toString) << "' for type '" << Red(FormatType.formatType(tpe)) << "'." << NewLine
+      vt << NewLine
+      vt << Code(loc, "no instance found") << NewLine
+      vt << NewLine
+      vt << Underline("Tip:") << " Add an instance for the type." << NewLine
```

Comment:
Quick comment: We could list the available instances. Thoughts?

Reply:
What happens if we have 100 instances? The flood might be inconvenient.

Reply:
I guess this is a problem for say `Eq` but less if you define your own type class.

Reply:
I think we leave it as is for now.